source/app/controllers/application_controller.rb

```diff
@@ -2,4 +2,25 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  helper_method :current_user, :logged_in?, :admin_user
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def logged_in?
+    unless !!current_user
```

Comment:
statements using unless are hard to reason about.
check this out: https://signalvnoise.com/posts/2699-making-sense-with-rubys-unless